sputnikdao2/src/proposals.rs

```diff
@@ -551,4 +559,16 @@ impl Contract {
             log!("Memo: {}", memo);
         }
     }
+
+    /// Act on given proposals, sequentially. The execution is interrupted and reverted if any of the proposal actions causes panic.
+    pub fn act_proposal_multi(&mut self, actions: Vec<ProposalAction>) {
+        actions.into_iter().for_each(|proposal| {
+            env::log_str(&format!(
+                "Processing action {} for proposal {}",
+                proposal.action.to_policy_label(),
+                proposal.id
+            ));
+            self.act_proposal(proposal.id, proposal.action, proposal.memo);
```

Comment:
there is an interesting issue here that act_proposal returns promises which may have callbacks attached as well. 
e.g. there potentially issues around callback trying to reverse state and failing due to complex interaction with other actions.

I think what make sense is to:
 -  add `finality: Option<bool>` to act_proposal, which if it's false, will not finalize the proposal automatically - so there is no execution involved.
 - from here call with `Some(false)` to not have any callback scheduled.
 - double check that finalize works for these proposals (right now I think it only works for expired and failed)

Reply:
This was addressed here, can you have a quick look and comment please?

https://github.com/near-daos/sputnik-dao-contract/pull/73/commits/4c53e7bac55795873f068c813ff240fc3970d401

Thanks.

Reply:
@TrevorJTClarke has this solution to the issue been accepted?